src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -314,6 +330,142 @@ class CreateTableStatementSegment(ansi.CreateTableStatementSegment):
     )
 
 
+class CreateUserStatementSegment(ansi.CreateUserStatementSegment):
+    """`CREATE USER` statement.
+
+    https://dev.mysql.com/doc/refman/8.0/en/create-user.html
+    """
+
+    _string_literal = OneOf(
+        Ref("QuotedLiteralSegment"), Ref("DoubleQuotedLiteralSegment")
+    )
+
+    _random_password = Sequence("RANDOM", "PASSWORD")
+    _auth_plugin = Ref("ObjectReferenceSegment")
+    _auth_string = _string_literal
```

Comment:
I'm not convinced it's worth the extra layer of abstraction to have these for one, or two references.

Any abstraction risks making the code more difficult to understand and I think doe these it would be clearer to just remove these and inline the values of them into the grammar.

For some of the other ones below, a level of abstraction and deduplication perhaps makes more sense (though even then I'm not convinced!) but think we should avoid overusing this IMHO as it does make reading the code more confusing in some senses as have to hope back and forth all over the place.

Inline comments are better for these IMHO if we want to make it clear that an `ObjectReferenceSegment` is an `_auth_plugin`?

Reply:
Yeah, you're right. I did it that way to organize as I was writing it, but it does read cleaner this way. https://github.com/sqlfluff/sqlfluff/pull/3289/commits/9e475bdb1e1683e9cf3bf150e103a1323df7a381